api-samples/nativeMessaging/app/manifest.json

```diff
@@ -0,0 +1,20 @@
+{
+  "manifest_version": 3,
+  "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDcBHwzDvyBQ6bDppkIs9MP4ksKqCMyXQ/A52JivHZKh4YO/9vJsT3oaYhSpDCE9RPocOEQvwsHsFReW2nUEc6OLLyoCFFxIb7KkLGsmfakkut/fFdNJYh0xOTbSN8YvLWcqph09XAY2Y/f0AL7vfO1cuCqtkMt8hFrBGWxDdf9CQIDAQAB",
+  "name": "Native Messaging Example",
+  "version": "1.0",
+  "description": "Send a message to a native application.",
+  "background": {
```

Comment:
main.js relies on `document.*` which isn't available in service workers. Since main.html launches as a popup, this can be used instead of any background/service worker instance so you can just remove this background portion.